esignet-integration-api/src/main/java/io/mosip/esignet/api/validator/FormatValidator.java

```diff
@@ -0,0 +1,41 @@
+package io.mosip.esignet.api.validator;
+
+import io.mosip.esignet.api.dto.AuthChallenge;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.List;
+import java.util.Map;
+
+
+@Component
+public class FormatValidator implements ConstraintValidator<Format, AuthChallenge> {
+    @Value("#{${mosip.esignet.supported-formats}}")
+    private Map<String, Object> supportedFormats;
+
+    @Override
+    public void initialize(Format constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(AuthChallenge authChallenge, ConstraintValidatorContext context) {
+        Object supportedFormatType = supportedFormats.get(authChallenge.getAuthFactorType());
```

Comment:
Can you reduce the nested if blocks here, and can we not consider that one authFactorType should be mapped to single format. In which cases are we expecting multiple formats?